### ISSUE TITLE
Redbean: edit zip file on emacs

### ISSUE THREAD

**Issue by renatoathaydes** [jart/cosmopolitan#932]
Hi @jart ! I really love Redbean, thanks for creating it!
I've used [something similar](https://renato.athaydes.com/posts/writing-your-own-smarthome-manager) (though definitely not actually portable) to create my smart home system!

Anyway, I wanted to be able to edit the redbean zip in emacs, but for some reason emacs is not recognizing it as a zip file.
Normally, emacs will open a buffer for zip files in "archive mode" which means it lets you see the contents of the zip and edit files in it as if it were normal files... but this is not possible with redbean.

I realize this is probably a problem with emacs, not Redbean! But I thought that it would be appropriate to ask you if you know what may be the issue given your extensive knowledge.
If I knew the root cause of the problem, I could try to provide a patch to the emacs project myself, but unfortunately I just don't have any idea how to get started on this.

Thanks for any assistance you may provide.

**Comment by pkulchenko** [comment]
Judging by the code in (arc-mode.el)[https://github.com/emacs-mirror/emacs/blob/master/lisp/arc-mode.el#L787], it searches for a ZIP signature from the beginning, but it should be done from the end of the file, so it finds something that looks like a signature (but is not) and likely bails out too soon. I don't think anything can be done here short of fixing it in emacs code. At least the updates themselves are done by external programs, so if the location of the zip is fixed, then modifications may work.

**Comment by renatoathaydes** [comment]
Well, just telling me where the function is that determines this already helps :) I will see if I can patch this to look at the end of the file as well.

**Comment by renatoathaydes** [comment]
I did some research... the zip file format is just problematic... I don't blame emacs for doing the "simple" thing.
See https://stackoverflow.com/questions/8593904/how-to-find-the-position-of-central-directory-in-a-zip-file

**Comment by pkulchenko** [comment]
I agree that it's not ideal, but there are some methods (search-backward and search-backward-regexp) that you can probably use to cobble something together similar to what's done in [archive-zip-summarize](https://github.com/emacs-mirror/emacs/blob/master/lisp/arc-mode.el#L2010) to find the start of the actual zip.
